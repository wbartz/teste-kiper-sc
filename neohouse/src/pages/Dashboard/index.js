import PropTypes from 'prop-types';
import React, { useContext, useState, useEffect } from 'react';
import { withRouter } from 'react-router-dom';
import { AppContext } from '../../containers/StoreProvider';
import Card from '../../components/Card';
import './index.scss';

const Dashboard = ({ getDashboard, dashboard, history }) => {
  const [hasData, setData] = useState(false);

  useEffect(() => {
    const getData = async () => {
      await getDashboard();
      setData(true);
    };
    if (!hasData) {
      getData();
    }
  }, [hasData, getDashboard]);

  return (
    <div className="page">
      <div className="dashboard-page">
        {dashboard ? (
          dashboard.map((line) => {
            return (
              <Card
                key={line.id}
                id={String(line.id)}
                title={line.name}
                footer={
                  <div className="content">
                    <div className="content-line">
                      <i className="material-icons">house</i>
                      <span>Apartamentos</span>
                      <span className="count">{line.apartments}</span>
                    </div>
                    <div className="content-line">
                      <i className="material-icons">person</i>
                      <span>Moradores</span>
                      <span className="count">{line.residents}</span>
                    </div>
                  </div>
                }
                icon="apartment"
                onEdit={() =>
                  history.push(`/apartamentos/${line.name.replace(' ', '-').toLowerCase()}`, {
                    block_id: line.id,
                  })
                }
              />
            );
          })
        ) : (
          <div className="empty">
            <i className="material-icons">apartment</i>
          </div>
        )}
      </div>
    </div>
  );
};

Dashboard.propTypes = {
  getDashboard: PropTypes.func.isRequired,
  history: PropTypes.shape({ push: PropTypes.func }).isRequired,
};

export default withRouter((props) => (
  <Dashboard {...useContext(AppContext)} {...props} />
));
