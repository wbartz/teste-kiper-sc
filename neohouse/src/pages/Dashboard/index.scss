@import '../../styles/variables';

.dashboard-page {
  margin-top: 70px;
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(306px, 1fr));
  grid-gap: 20px;

  .content {
    display: flex;
    justify-content: center;
    align-items: center;
    margin-top: 20px;
    width: 300px;

    .content-line {
      width: 100%;
      flex: 0.5;
      display: flex;
      align-items: center;
      flex-direction: column;
      height: 24px;

      span {
        flex: 0.5;
        text-align: center;
        font-size: $h6-font-size;
      }

      i.material-icons {
        color: $color-alternative;
        font-size: $h3-font-size;
      }
    }
  }
}
