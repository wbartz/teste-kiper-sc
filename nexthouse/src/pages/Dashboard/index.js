import PropTypes from 'prop-types';
import React, { useContext, useState, useEffect } from 'react';
import { AppContext } from '../../containers/StoreProvider';
import BlockIcon from '../../assets/ic-block.svg';

const Dashboard = ({ getDashboard }) => {
  const [hasData, setData] = useState(false);
  const [dashboard, setDashboard] = useState([]);

  useEffect(() => {
    if (!hasData) {
      getDashboard(setDashboard);
      setData(true);
    }
  }, [hasData]);

  return (
    <>
      {dashboard && dashboard.map(line => {
        return (
          <div className="card">
            <div className="icon">
              <img src={BlockIcon} alt={line.name} />
            </div>
            <div className="title">{line.name}</div>
            <div className="content">
              <div className="content-line">
                <img className="icon" alt="Apartamentos" />
                <span>{line.apartments}</span>
              </div>
              <div className="content-line">
                <img className="icon" alt="Moradores" />
                <span>{line.residents}</span>
              </div>
            </div>
          </div>
        )
      })}
    </>
  );
};

Dashboard.propTypes = {
  getDashboard: PropTypes.func.isRequired,
};

export default (props) => <Dashboard {...useContext(AppContext)} {...props} />;
