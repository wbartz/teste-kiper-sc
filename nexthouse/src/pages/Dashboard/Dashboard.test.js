import { shallow } from 'enzyme';
import React from 'react';
import Dashboard from '.';

describe('Dashboard', () => {
  const props = {
    getDashboard: jest.fn()
  }

  it('render without crashing', () => {
    expect(shallow(<Dashboard {...props} />)).toMatchSnapshot();
  });
});
