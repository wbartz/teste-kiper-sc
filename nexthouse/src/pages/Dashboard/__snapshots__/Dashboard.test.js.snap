// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard render without crashing 1`] = `ShallowWrapper {}`;
